tests_integration/test_snowpark_function.py

```diff
@@ -0,0 +1,417 @@
+                f"LIST @deployments"
+            )
+            filed_uploaded_in_this_test_case = [
+                staged_file
+                for staged_file in all_staged_files
+                if staged_file["name"]
+                .lower()
+                .startswith(f"deployments/{self.object_name_prefix.lower()}")
+            ]
+            return filed_uploaded_in_this_test_case
+        else:
+            return []
+
+    def clean_after_test_case(self) -> None:
+        self.delete_functions_created_in_this_test_case()
+        self.delete_files_uploaded_in_this_test_case()
+
+    def delete_functions_created_in_this_test_case(self) -> None:
+        functions_to_delete = self.query_functions_created_in_this_test_case()
+        for function in functions_to_delete:
+            name_with_args = re.sub(r" RETURN .*", "", function["arguments"])
+            self.sql_test_helper.execute_single_sql(f"DROP FUNCTION {name_with_args}")
+
+    def delete_files_uploaded_in_this_test_case(self) -> None:
+        files_to_remove = self.query_files_uploaded_in_this_test_case()
+        for file_to_remove in files_to_remove:
+            file_path_at_stage = file_to_remove["name"]
+            self.sql_test_helper.execute_single_sql(f"REMOVE @{file_path_at_stage}")
+
+
+class SnowparkFunctionTestSteps:
+    def __init__(self, setup: SnowparkFunctionTestSetup):
+        self._setup = setup
+
+    def snowpark_function_list_should_return_no_data(self) -> None:
+        result = self._setup.runner.invoke_with_config_and_integration_connection(
+            [
+                "snowpark",
+                "function",
+                "list",
+                "--like",
+                f"{self._setup.object_name_prefix}%",
+            ]
+        )
+        assert_that_result_is_successful_and_output_contains(result, "No data")
+
+    def snowpark_function_list_should_return_list_containing_function_at_the_first_place(
+        self, function_name: str, arguments: str, result_type: str
+    ) -> None:
+        result = self._setup.runner.invoke_with_config_and_integration_connection(
+            [
+                "--format",
+                "json",
+                "snowpark",
+                "function",
+                "list",
+                "--like",
+                f"{self._setup.object_name_prefix}%",
+            ]
+        )
+        assert_that_result_is_successful_and_json_output_contains_value_at_path(
+            result=result, path=[0, "name"], expected_value=function_name.upper()
+        )
+        assert_that_result_is_successful_and_json_output_contains_value_at_path(
+            result=result,
+            path=[0, "arguments"],
+            expected_value=f"{function_name}{arguments} RETURN {result_type}".upper(),
+        )
+        assert_that_result_is_successful_and_json_output_contains_value_at_path(
+            result=result, path=[0, "language"], expected_value="PYTHON"
+        )
```

Comment:
This method can be replaced with 

```
self._setup.runner.invoke_with_config_and_integration_connection(
            [
                "--format",
                "json",
                "snowpark",
                "function",
                "list",
                "--like",
                f"{self._setup.object_name_prefix}%",
            ]
        )

    assert contains_row_with(
        row_from_mock(mock_print),
        {
            "name": function_name.upper(),
            "arguments": f"{function_name}{arguments} RETURN {result_type}".upper(),
            "language": "PYTHON", 
        },
    )
```

WDYT?

Reply:
I don't like using mocks if we can execute an assertion on a real output. IMO it makes future refactors of printing more difficult. But as I wrote in another comment, I think we should talk about it.